src/components/visualisation-bar/index.js

```diff
@@ -0,0 +1,246 @@
+        ).add(
+            this.balanceElementsMap.get(label)
+                .add(this.balanceElementsMap.get(labelPc).add(this.balanceElementsMap.get(labelText)))
+        );
+    }
+
+    checkBuffer() {
+        const { buffer, value, max = this.options.value + this.options.buffer } = this.options;
+        const sum = buffer + value;
+        if (max && (sum > max)) {
+            this.buffer = max - value;
+        }
+    }
+
+    update() {
+        const { value: progress, buffer, max } = this.options;
+        this.__progress = Math.round((progress * 100) / this.max);
+        this.__buffer = Math.round((buffer * 100) / this.max);
+        this.__balance = max ? Math.round(100 - this.__buffer - this.__progress) : 0;
+        this.updateProgressFill();
+        this.updateBufferFill();
+        this.updateBalanceFill()
+        this.updateProgressLabel();
+        this.updateBufferLabel();
+        this.updateBalanceLabel();
+        this.updateCount();
+    }
+
+    updateProgressFill() {
+        this.progressElementsMap.get(wrapper).style('width', `${this.__progress}%`);
+        this.progressElementsMap.get(label).style('width', `${this.__progress}%`);
+        this.body.classed('without-progress', this.__progress === 0);
+    }
+
+    updateBufferFill() {
+        this.bufferElementsMap.get(wrapper).style('width', `${this.__buffer}%`);
+        this.bufferElementsMap.get(label).style('width', `${this.__buffer}%`);
+        this.body.classed('without-buffer', this.__buffer === 0);
+    }
+
+    updateBalanceFill() {
+        const { value, buffer, max } = this.options;
+        this.balanceElementsMap.get(wrapper).style('width', `${this.__balance ? this.__balance : 0}%`);
+        this.balanceElementsMap.get(progress).text((this.__balance ? max - value - buffer : '') || '');
+        this.balanceElementsMap.get(label).style('width', `${this.__balance}%`);
+        this.balanceElementsMap.get(fill).classed('inversed-text-color', true);
+        this.body.classed('without-balance', this.__balance === 0);
+    }
+
+    updateProgressLabel() {
+        if (this.options.withPercent) {
+            this.progressElementsMap.get(labelPc).text(this.__progress ? `(${this.__progress}%)` : '')
+        }
+    }
+
+    updateBufferLabel() {
+        if (this.options.withPercent) {
+            this.bufferElementsMap.get(labelPc).text(this.__buffer ? `(${this.__buffer}%)` : '')
+        }
+    }
+
+    updateBalanceLabel() {
+        if (this.options.withPercent) {
+            this.balanceElementsMap.get(labelPc).text(this.__balance ? `(${this.__balance}%)` : '')
+        }
+    }
+
+    updateCount() {
+        this.countElement.text(`${this.options.value}/${this.options.max || '-'}`);
+    }
+}
+
+
+export const visualisationBar = (options) => {
+    const selection = div();
+    new VisualisationBar(selection.node(), options);
+    return selection;
+}
```

Comment:
I can see quite a few linting issues here

You should be able to run `npm run lint -- --fix` to solve quite a few of them, it will list the remaining ones to fix for you